### ISSUE TITLE
Wrong number of output batches

### ISSUE THREAD

**Issue by adricarda** [NVIDIA/MinkowskiEngine#335]
Hello, 

starting from the pytorch lighnting example, I am trying to overfit some point clouds of a dataset. However, despite that the input batches is 2, the output of the network has a different number of batches, resulting in an error when computing the cross entropy loss function. 

**To Reproduce**
#%%
import torch
from torch import nn
import pytorch_lightning as pl
import numpy as np
import MinkowskiEngine as ME
import torch.nn.functional as F
import open3d as o3
import torch.utils.data as data
import glob
o3.set_verbosity_level(o3.VerbosityLevel.Error)

#%%

class DummyDataset(data.Dataset):
    def __init__(self, voxel_size):
        super().__init__()
        self.filenames = list(glob.glob("data/*.ply"))

    def __getitem__(self, i):
        filename = self.filenames[i]
        pcd = o3.read_point_cloud(filename)
        pcd = np.array(pcd.points, dtype=np.float32)

        pc_mean = pcd.mean(axis=0)
        pcd = pcd - pc_mean
        pc_L_max = np.max(np.sqrt(np.sum(abs(pcd ** 2), axis=-1)))
        pcd = pcd/pc_L_max

        random_labels = 0
        return {
            "coordinates": pcd,
            "labels": random_labels,
        }

    def __len__(self):
        return len(self.filenames)
#%%

def minkowski_collate_fn(list_data):
    r"""
    Collation function for MinkowskiEngine.SparseTensor that creates batched
    cooordinates given a list of dictionaries.
    """
    # print(list_data)
    coordinates = ME.utils.batched_coordinates(
        [d["coordinates"] for d in list_data],
        dtype=torch.float32,
    )
    features = torch.ones((len(coordinates), 1))
    labels = torch.tensor([d["labels"] for d in list_data])

    return {
        "coordinates": coordinates,
        "features": features,
        "labels": labels,
    }

#%%

class DummyNetwork(nn.Module):
    def __init__(self, in_channels, out_channels, D=3):
        nn.Module.__init__(self)
        self.net = nn.Sequential(
            ME.MinkowskiConvolution(in_channels, 32, 3, dimension=D),
            ME.MinkowskiBatchNorm(32),
            ME.MinkowskiReLU(),
            ME.MinkowskiConvolution(32, 64, 3, stride=2, dimension=D),
            ME.MinkowskiBatchNorm(64),
            ME.MinkowskiReLU(),
            ME.MinkowskiConvolutionTranspose(64, 32, 3, stride=2, dimension=D),
            ME.MinkowskiBatchNorm(32),
            ME.MinkowskiReLU(),
            ME.MinkowskiConvolution(32, out_channels, kernel_size=1, dimension=D),
        )

    def forward(self, x):
        return self.net(x)

class MinkowskiSegmentationModule(pl.LightningModule):
    r"""
    Segmentation Module for MinkowskiEngine.
    """

    def __init__(
        self,
        model,
        optimizer_name="SGD",
        lr=1e-3,
        weight_decay=1e-5,
        voxel_size=0.1,
        batch_size=2,
        val_batch_size=2,
        train_num_workers=0,
        val_num_workers=0,
    ):
        super().__init__()
        for name, value in vars().items():
            if name != "self":
                setattr(self, name, value)

        self.criterion = nn.CrossEntropyLoss()

    def train_dataloader(self):
        return data.DataLoader(
            DummyDataset(self.voxel_size),
            batch_size=self.batch_size,
            collate_fn=minkowski_collate_fn,
            shuffle=True,
        )

    def val_dataloader(self):
        return data.DataLoader(
            DummyDataset(self.voxel_size),
            batch_size=self.val_batch_size,
            collate_fn=minkowski_collate_fn,
            shuffle=True,
        )

    def forward(self, x):
        return self.model(x)

    def training_step(self, batch, batch_idx):
        coords = batch["coordinates"]
        feats = batch["features"]
        
        stensor = ME.SparseTensor(
            coordinates=coords, features=feats
        )
        # Must clear cache at regular interval
        if self.global_step % 10 == 0:
            torch.cuda.empty_cache()
        return self.criterion(self(stensor).F, batch["labels"].long())

    def validation_step(self, batch, batch_idx):
        coords = batch["coordinates"]
        feats = batch["features"]
        
        stensor = ME.SparseTensor(
            coordinates=coords, features=feats
        )
        out = self(stensor).F

        return self.criterion(out, batch["labels"].long())

    def configure_optimizers(self):
        return torch.optim.SGD(self.model.parameters(), lr=self.lr, weight_decay=self.weight_decay)
#%%

pl_module = MinkowskiSegmentationModule(DummyNetwork(1, 20, D=3), lr=0.0001)
trainer = pl.Trainer(max_epochs=1, gpus=1)
trainer.fit(pl_module)
# %%

**Desktop (please complete the following information):**

 - OS: [Ubuntu 20.04]
 - Python version: [3.7.0]
 - CUDA version: [11.2]
 - NVIDIA Driver version: [460]
 - Minkowski Engine version [0.5.0]

Is there something wrong with normalization maybe? From what I understand ME.sparseTensor converts coordinates to Int. Still, even removing normalization the error persists.

Thank you in advance.

**Comment by chrischoy** [comment]
Your code is based on https://github.com/NVIDIA/MinkowskiEngine/blob/master/examples/multigpu_lightning.py.
The reference example code runs perfectly fine without a problem.
